lib/middleware/csrfProtected.ts

```diff
@@ -0,0 +1,31 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { MiddlewareRequest, MiddlewareReturn } from "@lib/types";
+import { getCsrfToken } from "next-auth/client";
+
+export const csrfProtected = (protectedMethods: string[]): MiddlewareRequest => {
+  return async (req: NextApiRequest, res: NextApiResponse): Promise<MiddlewareReturn> => {
+    try {
+      if (isProtected(req, protectedMethods)) {
+        const csrfToken = await getCsrfToken({ req });
+        if (!csrfToken || csrfToken !== req.headers["x-csrf-token"]) {
```

Comment:
I'm curious as to why you decided to inverse the logic of the token check

from:  if(nextAuth is funtional and token is good) pass else block everything

to:  if(nextAuth is not functional or token is not good) block else pass everything

From a readability standpoint the inverse negation logic made the statement difficult for me to follow.  The default to pass everything that doesn't meet the statement criteria also seems like an anti-pattern for access control type logic.